waiter/src/waiter/core.clj

```diff
@@ -897,15 +901,15 @@
                                               service-description-builder kv-store service-id :effective? effective?)))
    :start-scheduler-syncer-fn (pc/fnk [[:settings [:health-check-config health-check-timeout-ms failed-check-threshold]]
                                        [:state clock user-agent-version]
-                                       service-id->password-fn* service-id->service-description-fn*]
+                                       scheduler-promise-chan service-id->password-fn* service-id->service-description-fn*]
                                 (let [http-client (hu/http-client-factory
                                                     {:client-name (str "waiter-syncer-" user-agent-version)
                                                      :conn-timeout health-check-timeout-ms
                                                      :socket-timeout health-check-timeout-ms
                                                      :user-agent (str "waiter-syncer/" user-agent-version)})
                                       available? (fn scheduler-available?
                                                    [scheduler-name service-instance service-description]
-                                                   (scheduler/available? service-id->password-fn* http-client
+                                                   (scheduler/available? service-id->password-fn* http-client scheduler-promise-chan
```

Comment:
Can we extract the scheduler from the channel inside the lambda and only pass the scheduler instance to scheduler/available??

```suggestion
                                                   (let [scheduler (async/<!! scheduler-promise-chan)]
                                                     (scheduler/available? service-id->password-fn* http-client scheduler
```

Reply:
We could, but that would mean adding another go-block (or a blocking call) here just to extract the scheduler. If you feel like that's a good idea then I can make the change.

Reply:
Since the scheduler is also the thing that invokes `available?`, I guess it's safe to assume that it's populated by the time this function is called. I can go ahead and switch it, and I'll add a comment to explain that the call shouldn't block.

Reply:
The value is already available when this function is invoked, we could even use a promise and throw an exception if it is invoked too eagerly. It removes the need to extract the scheduler from the consumer (`scheduler/available?`)

Reply:
Yes, I'd prefer to defer this change and wait to see if we actually end up needing it.